src/MusicPlaylistSecondary.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Layered implementations of secondary methods for {@code MusicPlaylist}.
+ */
+public abstract class MusicPlaylistSecondary implements MusicPlaylist {
+
+    /**
+     * MusicPlaylist object.
+     */
+    private MusicPlaylist playlist;
```

Comment:
You should not include an instance variables here. You must write your methods by using the kernel methods directly.